src/tiled/tilesetview.cpp

```diff
@@ -609,40 +968,52 @@ void TilesetView::mouseMoveEvent(QMouseEvent *event)
         return;
     }
 
-    if (!mEditTerrain) {
-        QTableView::mouseMoveEvent(event);
-        return;
-    }
+    if (mEditTerrain || mEditWangSet) {
+        //TODO this should tell the wangtemplate view to update
+        if (mEditWangSet && mWangSet) {
+            if (!mWangSet->wangIdIsValid(mWangId))
+                mWangId = 0;
+        }
 
-    const QPoint pos = event->pos();
-    const QModelIndex hoveredIndex = indexAt(pos);
-    int hoveredCorner = 0;
+        const QPoint pos = event->pos();
+        const QModelIndex hoveredIndex = indexAt(pos);
+        const QModelIndex previousHoveredIndex = mHoveredIndex;
+        mHoveredIndex = hoveredIndex;
+        int previousHoverCorner = mHoveredCorner;
 
-    if (hoveredIndex.isValid()) {
-        const QPoint center = visualRect(hoveredIndex).center();
+        if (mEditTerrain) {
+            int hoveredCorner = 0;
 
-        const auto t = tilesetGridTransform(*tilesetDocument()->tileset(), center);
-        const auto mappedPos = t.inverted().map(pos);
+            if (mHoveredIndex.isValid()) {
+                const QPoint center = visualRect(hoveredIndex).center();
 
-        if (mappedPos.x() > center.x())
-            hoveredCorner += 1;
-        if (mappedPos.y() > center.y())
-            hoveredCorner += 2;
-    }
+                const auto t = tilesetGridTransform(*tilesetDocument()->tileset(), center);
+                const auto mappedPos = t.inverted().map(pos);
 
-    if (mHoveredIndex != hoveredIndex || mHoveredCorner != hoveredCorner) {
-        const QModelIndex previousHoveredIndex = mHoveredIndex;
-        mHoveredIndex = hoveredIndex;
-        mHoveredCorner = hoveredCorner;
+                if (mappedPos.x() > center.x())
+                    hoveredCorner += 1;
+                if (mappedPos.y() > center.y())
+                    hoveredCorner += 2;
+
+                mHoveredCorner = hoveredCorner;
+            }
+        }
 
         if (previousHoveredIndex.isValid())
             update(previousHoveredIndex);
-        if (previousHoveredIndex != mHoveredIndex && mHoveredIndex.isValid())
+        if (mHoveredIndex.isValid() &&
+                (previousHoverCorner != mHoveredCorner || mHoveredIndex != previousHoveredIndex))
```

Comment:
In fact it updates the `previousHoverIndex` every time the mouse moves (and it's valid). I've changed it to the intended behavior.